src/v2/mod.rs

```diff
@@ -194,13 +195,48 @@ pub struct ChainParametersV1 {
     pub keys: types::UpdateKeysCollection<ChainParameterVersion1>,
 }
 
+#[derive(Debug, Clone)]
+/// Values of chain parameters that can be updated via chain updates.
+/// This applies to protocol version 6 and up.
+pub struct ChainParametersV2 {
+    /// Consensus protocol version 2 timeout parameters.
+    pub timeout_parameters: TimeoutParameters,
+    /// Minimum time interval between blocks.
+    pub min_block_time: Duration,
+    /// Maximum energy allowed per block.
+    pub block_energy_limit: Energy,
+    /// Euro per energy exchange rate.
+    pub euro_per_energy: ExchangeRate,
+    /// Micro ccd per euro exchange rate.
+    pub micro_ccd_per_euro: ExchangeRate,
+    pub cooldown_parameters: CooldownParameters,
+    pub time_parameters: TimeParameters,
+    /// The limit for the number of account creations in a block.
+    pub account_creation_limit: CredentialsPerBlockLimit,
+    /// Parameters related to the distribution of newly minted CCD.
+    pub mint_distribution: MintDistributionV1,
+    /// Parameters related to the distribution of transaction fees.
+    pub transaction_fee_distribution: TransactionFeeDistribution,
+    /// Parameters related to the distribution of the GAS account.
+    pub gas_rewards: GASRewardsCPV2,
```

Comment:
I did not notice this before, but since we dont' use CPV2 in naming of other structs (we just use V?) then we should try not to do that with GasRewards either.

And I think we should name this `GasRewards` rather than `GAS`.

Reply:
So the `CPV2` suffix stems from the corresponding type in the proto files, but I agree that it is different from the current convention in the SDK, so I'll change it.

Reply:
Regarding using `Gas` instead of `GAS`, the version 0 of gas rewards is called `GASRewards`, so I kept it the same here.
I wouldn't mind changing it though

Reply:
Right. Let's keep the GAS then